src/test/java/edu/iis/powp/gui/TestPlotterApp.java

```diff
@@ -21,106 +31,137 @@
 
-	}
+    }
 
-	/**
-	 * Setup driver manager, and set default IPlotter for application.
-	 * 
+    /**
+     * Setup driver manager, and set default IPlotter for application.
+     *
 	 * @param application
 	 *            Application context.
-	 */
-	private static void setupDrivers(Application application) {
-		IPlotter clientPlotter = new ClientPlotter();
-		application.addDriver("Client Plotter", clientPlotter);
-
-		DrawPanelController drawerController = DrawerFeature.getDrawerController();
-		IPlotter plotter = new LineAdapterPlotterDriver(drawerController, LineFactory.getBasicLine(), "basic");
-		application.addDriver("Line Simulator", plotter);
-		application.getDriverManager().setCurrentPlotter(plotter);
-
-		plotter = new LineAdapterPlotterDriver(drawerController, LineFactory.getSpecialLine(), "special");
-		application.addDriver("Special line Simulator", plotter);
-		application.updateDriverInfo();
-	}
-
-	private static void setupWindows(Application application) {
-
-		CommandManagerWindow commandManager = new CommandManagerWindow(CommandsFeature.getPlotterCommandManager());
-		application.addWindowComponent("Command Manager", commandManager);
-
-		CommandManagerWindowCommandChangeObserver windowObserver = new CommandManagerWindowCommandChangeObserver(
-				commandManager);
-		CommandsFeature.getPlotterCommandManager().getChangePublisher().addSubscriber(windowObserver);
-	}
-
-	/**
-	 * Setup menu for adjusting logging settings.
-	 * 
+     */
+    private static void setupDrivers(Application application) {
+        IPlotter clientPlotter = new ClientPlotter();
+        application.addDriver("Client Plotter", clientPlotter);
+
+        DrawPanelController drawerController = DrawerFeature.getDrawerController();
+        IPlotter plotter = new LineAdapterPlotterDriver(drawerController, LineFactory.getBasicLine(), "basic");
+        application.addDriver("Line Simulator", plotter);
+        application.getDriverManager().setCurrentPlotter(plotter);
+
+        plotter = new LineAdapterPlotterDriver(drawerController, LineFactory.getSpecialLine(), "special");
+        application.addDriver("Special line Simulator", plotter);
+        application.updateDriverInfo();
+    }
+
+    private static void setupWindows(Application application) {
+
+        CommandManagerWindow commandManager = new CommandManagerWindow(CommandsFeature.getPlotterCommandManager());
+        application.addWindowComponent("Command Manager", commandManager);
+
+        CommandManagerWindowCommandChangeObserver windowObserver = new CommandManagerWindowCommandChangeObserver(
+                commandManager);
+        CommandsFeature.getPlotterCommandManager().getChangePublisher().addSubscriber(windowObserver);
+
+        CommandRegistry commandRegistry = new CommandRegistry();
+        commandRegistry.addComplexCommandRegisterActionListener(new ActionListener() {
```

Comment:
FIX TO MERGE: Great deal of this code can be moved outside TestPlotterApp (only called within). 
The goal is to seperate responsibilites and make merging for other groups easier.